src/shared/Core/Trace2.cs

```diff
@@ -0,0 +1,278 @@
+    /// </summary>
+    /// <param name="gcmVersion">The version of GCM.</param>
+    /// <param name="eventFormatVersion">
+    /// The EVENT format version, which is defined in Git.
+    /// This may need to be incremented in the future to align with updated versions in Git.
+    /// </param>
+    /// <param name="filePath">Path of the file this method is called from.</param>
+    /// <param name="lineNumber">Line number of file this method is called from.</param>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed")]
+    void WriteVersion(
+        string gcmVersion,
+        string eventFormatVersion = "3",
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0);
+
+    /// <summary>
+    /// Writes argv and time information associated with application start to
+    /// trace writer.
+    /// </summary>
+    /// <param name="argv">Exe name and arguments.</param>
+    /// <param name="timeInMilliseconds">Elapsed time in milliseconds since GCM started.</param>
+    /// <param name="filePath">Path of the file this method is called from.</param>
+    /// <param name="lineNumber">Line number of file this method is called from.</param>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed")]
+    void WriteStart(
+        IList<string> argv,
+        double timeInMilliseconds,
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0);
+    
+    /// <summary>
+    /// Writes exit code and execution time to trace writer.
+    /// </summary>
+    /// <param name="code">Application exit code.</param>
+    /// <param name="timeInMilliseconds">Elapsed time in milliseconds since GCM started.</param>
+    /// <param name="filePath">Path of the file this method is called from.</param>
+    /// <param name="lineNumber">Line number of file this method is called from.</param>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed")]
+    void WriteExit(
+        int code,
+        double timeInMilliseconds,
+        [System.Runtime.CompilerServices.CallerFilePath]
+        string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber]
+        int lineNumber = 0);
+}
+
+public class Trace2: DisposableObject, ITrace2
+{
+    private const string ParentSidVariable = "GIT_TRACE2_PARENT_SID";
+    private const string CurrentSidVariable = "GCM_TRACE2_PARENT_SID";
+    private readonly ICommandContext _context;
+    private bool _isDisposed;
+    private CollectorWriter _writer;
+
+    public Trace2(ICommandContext context)
+    {
+        _context = context;
+    }
+
+    public string GetSid()
+    {
```

Comment:
If I'm reading this right, this will create a new SID every time it is called.  And it'll keep getting longer (because of the setenv at the bottom will cause the getenv on the next call to get the updated value).

Shouldn't this have a static variable hidden within it and just return it for subsequent calls?

Reply:
Good catch, thank you!